src/TemplateManager.php

```diff
@@ -28,6 +29,11 @@ class TemplateManager
      */
     protected $eventManager;
 
+    /**
+     * @var EventFactoryInterface
+     */
+    protected $eventFactory;

```

Comment:
These event-related properties are not needed if you `use` the `EventConsumerTrait`. Not only does it give you the necessary protected getters and setters for them as well, it also provides the shorthand helper methods `_attach()`, `_trigger()` and `_detach()`.